### ISSUE TITLE
History Viewer: Toggle unpublished versions

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-versioned#105]
# Acceptance Criteria

 * The version list shows published versions by default
 * I can optionally toggle on listing of unpublished versions
 * I get a clear indication if the list is filtered to published versions only

# Notes

 * Designs at https://projects.invisionapp.com/share/FJEHOLLEN#/screens/263513634_Pages_History_Tab
 * Preference does not need to persist between page views or user sessions
 * Uses same popover as the [compare mode toggle](https://github.com/silverstripe/silverstripe-versioned/issues/96)

**Comment by chillu** [comment]
> I get a clear indication if the list is filtered to published versions only

@clarkepaul Do you think that's necessary? At the moment, it would be quite easy to miss for users that they're viewing a filtered list by default (only published versions). Presumably that's to avoid too much clutter? Maybe it's overkill, given that we have pagination?

**Comment by clarkepaul** [comment]
I think prioritising the list by published items is what people would typically want and aligns more to the actual history of the live page. I would think the list could get bloated quite easily if someone is saving every few minutes but this is one of things which is hard to have clear stats on as every page would differ as well as an editor publishing process.

From our previous discussion I think you are just going to show draft versions in the mix of published items to save dev time , and I'm cool with that. This means that the "Compare mode" toggle is also coming out of the dropdown so that the dropdown isn't required at all for now.

**Comment by chillu** [comment]
@clarkepaul I think the main value of showing unpublished versions is allowing users to compare the *latest* unpublished version to the *latest* published one (so review what's about to be published). The use case for viewing older unpublished versions isn't that strong, although it's required for a full audit record. Author A might have edited a record a few times without publishing, then Author B hits the publish button. You should still be able to attribute those edits to Author A.

**Comment by clarkepaul** [comment]
So we currently have it set up in pages that when there is a most recent draft/modified version the history view pre-checks the "Show draft versions" so that it appears at the top, and when the most recent version is a published version draft/modified items are hidden. Although this helps with your scenario I think a better way to handle that will be showing the recent activity since last published on the edit view (then they don't need to navigate to history to be informed).

**Comment by raissanorth** [comment]
> Preference does not need to persist between page views or user sessions

@chillu Can you please elaborate on this? I don't quite know what this is supposed to convey. Thanks.

**Comment by robbieaverill** [comment]
It's worth mentioning that we probably won't be able to use React state to filter the existing list of versions that we have with this toggle - it'll need to add another filter to the GraphQL query in some way, since the query is paginated

**Comment by clarkepaul** [comment]
Had a quick chat to @raissanorth about the AC "I get a clear indication if the list is filtered to published versions only" and I don't have great solution to this to make it super obvious to users except the showing the status of published items. I think we can go ahead with the current design and see if this presents its self as an actual issue with users, as I think the risk is fairly low.

@chillu I would like the functionality to change slightly from what exists on page history currently, I think we should only show published items in the list but include the most recent version even if its DRAFT or MODIFIED. 
I'm adding an AC that no matter what the most recent version is that it always shows.

**Comment by chillu** [comment]
>  I'm adding an AC that no matter what the most recent version is that it always shows.

Good point, thanks for adding that. It sounds like this story will be on the backburner for a while anyway :/

**Comment by robbieaverill** [comment]
This would be unblocked by #182

**Comment by chillu** [comment]
Removing from Zenhub epic, since that's closed now (no more funding through that epic)
